src/main/scala/com/ubirch/util/elasticsearch/EsBulkClient.scala

```diff
@@ -49,7 +49,7 @@ trait EsBulkClientBase extends StrictLogging {
     def afterBulk(executionId: Long, request: BulkRequest, failure: Throwable): Unit = {
       logger.error(s"afterBulk($executionId, number of actions: #${request.numberOfActions()}, ${request.estimatedSizeInBytes()}, trying it once more)", failure)
 
-      esClient.bulkAsync(request, RequestOptions.DEFAULT, new ActionListener[BulkResponse]() {
+      bulkAsyncAsJava.accept(request, new ActionListener[BulkResponse]() {
```

Comment:
This definitely makes the code better. Just as a comment. I am just wondering if there is conversion out there that makes the implicit conversation between java lambdas and scala lambdas.m
But this is optional here. :)

Reply:
The other question for the next version, how we could handle an N amount of retries with this elastic search listeners.

Reply:
I have never thought about the lambda conversions. Will research on it. And yes, I could open a ticket for the N amount of retries for the bulk processing, but I'm wondering, if we should focus on/ prioritise this, when one retry would be enough and there are many other important tasks.. ;) I'm a bit torn between the two positions. 
Tests regarding the retries I should definitly add though. Thanks for reminding!